src/test/java/com/google/cloud/genomics/utils/ShardBoundaryTest.java

```diff
@@ -75,21 +88,29 @@ public void testGetStrictReadPredicate() {
       Read[] reads = new Read[] { overlapStartWithinExtent, overlapStartExtent, atStartWithinExtent,
               atStartOverlapExtent, beyondStartWithinExtent, beyondOverlapExtent };
 
-      Predicate<Read> shardPredicate = ShardBoundary.getStrictReadPredicate(start);
+      Predicate<Read> shardPredicate = ShardBoundary.getStrictReadPredicate(start, null);
       List<Read> filteredReads = Lists.newArrayList(Iterables.filter(Arrays.asList(reads), shardPredicate));
       assertEquals(4, filteredReads.size());
       assertThat(filteredReads, CoreMatchers.allOf(CoreMatchers.hasItems(atStartWithinExtent,
           atStartOverlapExtent, beyondStartWithinExtent, beyondOverlapExtent)));
   }
 
   @Test
+  public void testGetStrictReadPredicateInsufficientFields() {
+    thrown.expect(IllegalArgumentException.class);
+    thrown.expectMessage(containsString("Insufficient fields requested in partial response. "
+        + "At a minimum include 'alignments(alignment)' to enforce a strict shard boundary."));
+    ShardBoundary.getStrictReadPredicate(123, "alignments(alignedSequence)");
+  }
+
+  @Test
   public void testGetNonVariantOverlapsPredicate() {
     long start = 1000L;
 
     Variant overlapStart = Variant.newBuilder().setReferenceBases("T").addAlternateBases("A").setStart(900L).build();
     Variant overlapStartNonVariant = Variant.newBuilder().setReferenceBases("T").setStart(900L).setEnd(1005L).build();
 
-    Predicate<Variant> shardPredicate = ShardBoundary.getNonVariantOverlapsPredicate(start);
+    Predicate<Variant> shardPredicate = ShardBoundary.getNonVariantOverlapsPredicate(start, null);
```

Comment:
This is good, but there is no test when the fields are provided as an non-null argument.  The function is an OR statement based on the following so this predicate would be true if for fields the start is within the bound and matches `VARIANT_FIELD_PATTERN`:
```Java
return Predicates.or(VariantUtils.IS_NON_VARIANT_SEGMENT, getStrictVariantPredicate(start, fields));
```